packages/enzyme/src/Debug.js

```diff
@@ -12,13 +11,12 @@ import {
   propsOfNode,
   childrenOfNode,
 } from './RSTTraversal';
+import { getAdapter } from './Utils';
 
 const booleanValue = Function.bind.call(Function.call, Boolean.prototype.valueOf);
 
-export function typeName(node) {
-  return typeof node.type === 'function'
```

Comment:
i think all this logic needs to stay as a fallback for when the adapter lacks `.displayNameOfNode`